java/test/com/google/jglome/GlomeTest.java

```diff
@@ -0,0 +1,177 @@
+import java.security.InvalidKeyException;
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Contains tests for com.google.jglome.Glome functionality.
+ */
+public class GlomeTest {
+
+  public static class KeyPair {
+
+    private byte[] publicKey;
+    private byte[] privateKey;
+
+    KeyPair(byte[] publicKey, byte[] privateKey) {
+      this.publicKey = publicKey;
+      this.privateKey = privateKey;
+    }
+
+    public byte[] getPrivateKey() {
+      return privateKey;
+    }
+
+    public byte[] getPublicKey() {
+      return publicKey;
+    }
+  }
+
+  final static int N_TEST_VECTORS = TEST_VECTORS.size();
+
+  Glome[][] glomeManagers = new Glome[N_TEST_VECTORS][2]; // first is for A, second is for B
+  KeyPair[] aKeys = new KeyPair[N_TEST_VECTORS];
+  KeyPair[] bKeys = new KeyPair[N_TEST_VECTORS];
+
+  GlomeTest() throws MinPeerTagLengthOutOfBoundsException, InvalidKeyException {
+    for (int i = 0; i < N_TEST_VECTORS; i++) {
+      TestVector testVector = TEST_VECTORS.get(i);
+      aKeys[i] = new KeyPair(testVector.getKa(), testVector.getKah());
+      bKeys[i] = new KeyPair(testVector.getKb(), testVector.getKbh());
+      glomeManagers[i][0] = new GlomeBuilder(bKeys[i].getPublicKey(), 32)
+          .setPrivateKey(aKeys[i].getPrivateKey())
+          .build();
+      glomeManagers[i][1] = new GlomeBuilder(aKeys[i].getPublicKey(), 28)
+          .setPrivateKey(bKeys[i].getPrivateKey())
+          .build();
+    }
+  }
+
+  @Test
+  public void testShouldFailWhenMinPeerTagLengthIsOutOfBounds() {
+    int[] minPeerTagLength = new int[]{MIN_TAG_LENGTH - 1, MAX_TAG_LENGTH + 1};
+
+    for (int len : minPeerTagLength) {
+      try {
+        new GlomeBuilder(aKeys[0].getPublicKey(), len);
+      } catch (MinPeerTagLengthOutOfBoundsException e) {
+        assertEquals(e.getMessage(),
+            String.format("minPeerTagLength argument should be in [%d..%d] range. Got %d.",
+                MIN_TAG_LENGTH, MAX_TAG_LENGTH, len));
+      }
+    }
+  }
+
+  @Test
+  public void checkCorrectMinPeerTagLength() {
+    for (int len = MIN_TAG_LENGTH; len <= MAX_TAG_LENGTH; len++) {
+      try {
+        new GlomeBuilder(aKeys[0].getPublicKey(), len);
+      } catch (MinPeerTagLengthOutOfBoundsException e) {
+        assertEquals(e.getMessage(),
+            String.format("minPeerTagLength argument should be in [%d..%d] range. Got %d.",
+                MIN_TAG_LENGTH, MAX_TAG_LENGTH, len));
+      }
+    }
+  }
+
+  @Test
+  public void testShouldFailWhenCounterIsOutOfBounds() {
+    TestVector vector = TEST_VECTORS.get(0);
+    int[] counters = new int[]{MIN_CNT_VALUE - 1, MAX_CNT_VALUE + 1};
+
+    for (int cnt : counters) {
+      try {
+        glomeManagers[0][0].generateTag(vector.getMsg(), cnt);
+      } catch (CounterOutOfBoundsException e) {
+        assertEquals(e.getMessage(),
+            String.format("Counter should be in [%d..%d] range. Got %d.", MIN_CNT_VALUE,
+                MAX_CNT_VALUE, cnt));
+      }
```

Comment:
Again, the test should not pass if no exception is thrown.

Reply:
Fixed